src/pt/pinkseaunicorn/src/eu/kanade/tachiyomi/extension/pt/pinkseaunicorn/PinkSeaUnicorn.kt

```diff
@@ -16,7 +19,19 @@ class PinkSeaUnicorn : Madara(
 
     override val client: OkHttpClient = super.client.newBuilder()
         .rateLimit(1, 2, TimeUnit.SECONDS)
+        .addNetworkInterceptor(::checkPasswordProtectedIntercept)
         .build()
 
+    private fun checkPasswordProtectedIntercept(chain: Interceptor.Chain): Response {
+        val response = chain.proceed(chain.request())
+
+        if (response.request.url.queryParameter("password-protected") != null) {
```

Comment:
I also said to change this, as network interceptor gets the redirect url in the request as well
```
if (chain.request().url.queryParameter("password-protected") != null) {
```

Reply:
> I also said to change this, as network interceptor gets the redirect url in the request as well
> 
> ```
> if (chain.request().url.queryParameter("password-protected") != null) {
> ```

Sorry I didn't see the last message
will change it right away